landlab/components/overland_flow/generate_overland_flow_deAlmeida.py

```diff
@@ -380,11 +381,14 @@ def rainfall_intensity(self):
         return self._rainfall_intensity
 
     @rainfall_intensity.setter
-    def rainfall_intensity(self, rainfall_intensity):
-        if rainfall_intensity >= 0:
-            self._rainfall_intensity = rainfall_intensity
+    def rainfall_intensity(self, new_val):
+        message = "Rainfall intensity must be positive"
+        if isinstance(new_val, float):
+            assert new_val >= 0, message
         else:
-            raise ValueError("Rainfall intensity must be positive")
+            assert (new_val >= 0).all(), message
```

Comment:
Same as the above regarding the *assert*. Instead:
```python
if np.any(new_val < 0.0):
    raise ValueError("Rainfall intensity must be positive")
```
This should handle both scalar and array intensities.